src/App.js

```diff
@@ -1,23 +1,85 @@
-import logo from './logo.svg';
-import './App.css';
+import { useState } from 'react';
+import PageHeader from './components/PageHeader';
+import PageMain from './components/PageMain';
+import PageFooter from './components/PageFooter';
+import SearchForm from './components/SearchForm';
+import MovieDetails from './components/MovieDetails';
+import MoviesList from './components/MoviesList';
+import MoviesModal from './components/MoviesModal';
+import ErrorBoundary from './utils/ErrorBoundary';
+import { ReactComponent as SearchIcon } from './images/icons/search.svg';
+import './styles/App.scss';
 
 function App() {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [isMovieDetailsMode, setIsMovieDetailsMode] = useState(false);
+  const [movieDetails, setMovieDetails] = useState({ type: 'add', data: {} });
+
+  const showMovieModal = (type, movieData) => {
+    setMovieDetails({ type, data: movieData });
+    setIsModalVisible(true);
+  };
+
+  const handleHeaderBtnClick = () => {
+    return isMovieDetailsMode
+      ? setIsMovieDetailsMode(false)
+      : showMovieModal('add');
+  };
+
+  const handleMovieClick = movie => {
+    setIsMovieDetailsMode(true);
+    setMovieDetails({ ...movieDetails, data: movie });
+  };
+
+  const headerBtn = (
+    <button
+      type="button"
+      className={`${isMovieDetailsMode ? 'btn-link' : 'btn'}`}
+      onClick={handleHeaderBtnClick}
+    >
+      {isMovieDetailsMode ? <SearchIcon /> : <span>+ Add movie</span>}
+    </button>
+  );
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
+    <div className="page">
+      <ErrorBoundary>
```

Comment:
You can have just one ErrorBoundary but at the top of your component tree

Reply:
Fixed